k8s/webhooks.go

```diff
@@ -299,6 +325,82 @@ func (w *WebhookServer) HandleMutateHTTP(writer http.ResponseWriter, req *http.R
 	writer.Write(bytes)
 }
 
+// HandleConvertHTTP is the HTTP HandlerFunc for a kubernetes CRD conversion webhook call
+// nolint:errcheck,revive,funlen
+func (w *WebhookServer) HandleConvertHTTP(writer http.ResponseWriter, req *http.Request) {
+	// Only POST is allowed
+	if req.Method != http.MethodPost {
+		writer.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	}
+
+	// Read the body
+	body, err := io.ReadAll(req.Body)
+	defer req.Body.Close()
+	if err != nil {
+		writer.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	// Unmarshal the ConversionReview
+	rev := conversion.ConversionReview{}
+	err = json.Unmarshal(body, &rev)
+	if err != nil {
+		writer.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	if rev.Response == nil {
+		rev.Response = &conversion.ConversionResponse{
+			ConvertedObjects: make([]runtime.RawExtension, 0),
+		}
+	}
+	rev.Response.UID = rev.Request.UID
+
+	// Go through each object in the request
+	for _, obj := range rev.Request.Objects {
+		// Partly unmarshal to find the kind and APIVersion
+		tm := metav1.TypeMeta{}
+		err = json.Unmarshal(obj.Raw, &tm)
+		if err != nil {
+			writer.WriteHeader(http.StatusBadRequest)
+			return
+		}
+		// Get the associated converter for this kind
+		conv, ok := w.converters[gk(tm.GroupVersionKind().Group, tm.Kind)]
+		if !ok {
+			// No converter for this kind
+			writer.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		// Do the conversion
+		// Partial unmarshal to get kind and APIVersion
+		res, err := conv.Convert(RawKind{
+			Kind:       tm.Kind,
+			APIVersion: tm.APIVersion,
+			Group:      tm.GroupVersionKind().Group,
+			Version:    tm.GroupVersionKind().Version,
+			Raw:        obj.Raw,
+		}, rev.Request.DesiredAPIVersion)
+		if err != nil {
+			// Conversion error
+			writer.WriteHeader(http.StatusInternalServerError)
+			return
+		}
+		rev.Response.ConvertedObjects = append(rev.Response.ConvertedObjects, runtime.RawExtension{
+			Raw: res,
+		})
+	}
```

Comment:
I don't think we should be returning non-200 responses here, instead we should use the `Response.Result` accordingly, since at this point all errors are related to the "business logic", not the transport protocol.

I suggest copying the logic from [the Kubernetes conversion test](https://github.com/kubernetes/kubernetes/blob/master/test/images/agnhost/crd-conversion-webhook/converter/framework.go#L88-L115) into here.